### PR TITLE
virtualise test entrypoint

### PR DESCRIPTION
**Pull request by mansona**

### CODE REVIEW

File: packages/core/src/virtual-test-entrypoint.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    const resolvedValues = await Promise.all(Object.values(fastbootModules));
+
+    Object.keys(fastbootModules).forEach((k, i) => {
+      d(k, function(){ return resolvedValues[i];});
+    })
+  }
+{{/if}}
+
+
+{{#if lazyRoutes}}
```
Comment (ef4) [inline packages/core/src/virtual-test-entrypoint.ts:123]: I think a test entrypoint would never use this.

File: packages/core/src/virtual-test-entrypoint.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+  for (let relativePath of appFiles.tests) {
+    amdModules.push(importPaths(resolver, appFiles, relativePath));
+  }
+
+  let src = entryTemplate({
+    amdModules,
+    testSuffix: true,
+    // this is a backward-compatibility feature: addons can force inclusion of test support modules.
+    defineModulesFrom: './-embroider-implicit-test-modules.js',
```
Comment (ef4) [inline packages/core/src/virtual-test-entrypoint.ts:68]: Should the tests entrypoint be doing this? Remember that it imports the app entrypoint, so anything the app entrypoint is doing shouldn't be repeated in here.

**Review by ef4 — COMMENTED** [review]
I think this either needs to be radically reduced down to things that only happen in the tests entrypoint or (if we really do need a lot of this code) needs to share the implementation with the app entrypoint. I don't want to just duplicate the implementation.

File: packages/core/src/virtual-test-entrypoint.ts (hunk 1 above)
Comment (mansona) [reply packages/core/src/virtual-test-entrypoint.ts]: removed 👍

File: packages/core/src/virtual-test-entrypoint.ts (hunk 2 above)
Comment (mansona) [reply packages/core/src/virtual-test-entrypoint.ts]: you're right this shouldn't be needed, it's already importing the app entrypoint and that is handled there 👍 removed

**Review by ef4 — APPROVED** [review]
Much smaller!

**Pull request merged**